Servers/UI/OJS.Servers.Ui/ClientApp/src/redux/services/admin/submissionsAdminService.ts

```diff
@@ -1,71 +1,42 @@
     ISubmissionsAdminGridViewType,
 } from '../../../common/types';
 import { IRetestSubmissionUrlParams } from '../../../common/url-types';
+import getCustomBaseQuery from '../../middlewares/customBaseQuery';
 
 export const submissionsAdminService = createApi({
-    reducerPath: 'adminSubmissionsService',
-    baseQuery: fetchBaseQuery({
-        credentials: 'include',
-        baseUrl: `${import.meta.env.VITE_ADMINISTRATION_URL}/${defaultPathIdentifier}/submissions`,
-        prepareHeaders: (headers) => {
-            headers.set('Content-Type', 'application/json');
-            return headers;
-        },
-        responseHandler: async (response: Response) => {
-            const contentType = response.headers.get('Content-Type');
-            if (contentType?.includes('application/octet-stream')) {
-                const contentDisposition = response.headers.get('Content-Disposition');
-                let filename = 'submission.zip';
-                if (contentDisposition) {
-                    const match = contentDisposition.match(/filename="?(.+?)"?(;|$)/);
-                    if (match) {
-                        filename = decodeURIComponent(match[1]);
-                    }
-                }
-                const blob = await response.blob();
-                return { blob, filename };
-            }
-
-            if (response.headers.get('Content-Length')) {
-                return '';
-            }
-
-            return response.json();
-        },
-    }),
+    reducerPath: 'submissionsAdminService',
+    baseQuery: getCustomBaseQuery('submissions'),
     endpoints: (builder) => ({
         getAllSubmissions: builder.query<IPagedResultType<ISubmissionsAdminGridViewType>, IGetAllAdminParams>({
-            query: ({
-                filter, page, itemsPerPage, sorting }) => ({
+            query: ({ filter, page, itemsPerPage, sorting }) => ({
                 url: '/getAll',
                 params: {
                     filter,
                     page,
                     itemsPerPage,
                     sorting,
-                } }) }),
+                },
+            }),
+        }),
         deleteSubmission: builder.mutation({
             query: (id) => ({
                 url: `/delete/${id}`,
                 method: 'DELETE',
-            }) }),
+            }),
+        }),
         retest: builder.mutation({
             query: (submissionId) => ({
                 url: `/retest/${submissionId}`,
                 method: 'POST',
-            }) }),
-        downloadFileSubmission: builder.query<{ blob: Blob; filename: string }, IRetestSubmissionUrlParams>({
-            query: ({ id }) => ({
-                url: `/download/${id}`,
-            }) }),
+            }),
+        }),
+        // eslint-disable-next-line max-len
+        downloadFileSubmission: builder.query<{blob: Blob; filename: string }, IRetestSubmissionUrlParams>({ query: ({ id }) => ({ url: `/Download/${id}` }) }),
```

Comment:
Can you split on 2 or more lines instead of turning off the rule

Reply:
No , there is fight between 2 rules, one says that it is too long for one line, the other says that it must be on one line